alpha/ui.cpp

```diff
@@ -193,3 +206,61 @@ void _TS_UI::task(void* parameter)
     TS_HAL.sleep(TS_SleepMode::Task, 20);
   }
 }
+
+void _TS_UI::draw_battery_icon()
+{
+  if (TS_HAL.power_is_charging())
+  {
+    TS_HAL.lcd_drawbitmap(
+      TS_LCD_WIDTH - TS_LCD_BORDER - ICON_BATT_W,
+      TS_LCD_BORDER,
+      ICON_BATT_W,
+      ICON_BATT_H,
+      icon_batt_c
+    );
+  }
+  else
+  {
+    uint8_t batt = TS_HAL.power_get_batt_level();
```

Comment:
this involves a call to i2c, we should add a local cache that only allows a pull after 10s or so

Reply:
I guess you also need to get charging status and a charging icon